src/main/scala/PacketAssembler/disassembler.scala

```diff
@@ -119,16 +119,22 @@ class PacketDisAssembler extends Module {
   val counter_byte = RegInit(0.U(3.W)) //counter for bits in bytes
 
   //packet status
+  val done = RegInit(false.B)
+
   val pdu_length = RegInit(0.U(8.W))
   val pdu_length_valid = RegInit(false.B)
   val pdu_length_ready = RegInit(false.B)
-  val done = RegInit(false.B)
+  val pdu_length_fire = RegInit(false.B)
+
   val flag_aa = RegInit(true.B)
   val flag_aa_valid = RegInit(false.B)
   val flag_aa_ready = RegInit(false.B)
```

Comment:
Keep in mind that using a register for ready signals delays it a clock cycle, which will mean valid will stay high for an extra cycle. I would use a Wire instead, or just use the input directly